src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -68,16 +76,37 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzExtTreeItem[]> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        const listOfDatabases: DatabaseListResult = await client.databases.listByServer(this.resourceGroup, this.name);
-        return this.createTreeItemsWithErrorHandling(
-            listOfDatabases,
-            'invalidPostgresServer',
-            (database) => {
-                return database.name && !['azure_maintenance', 'azure_sys'].includes(database.name) ? new PostgresDatabaseTreeItem(this, database.name) : undefined;
-            },
-            (database) => database.name
-        );
+        if (!(this.connectionString.username && this.connectionString.password)) {
+            const { username, password } = await this.getCredentials();
+            if (username && password) {
+                await this.updateConnectionString(username, password);
+            }
+        }
+        if (this._azureId) {
+            const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+            const listOfDatabases: DatabaseListResult = await client.databases.listByServer(nonNullProp(this, 'resourceGroup'), nonNullProp(this, 'azureName'));
+            return this.createTreeItemsWithErrorHandling(
+                listOfDatabases,
+                'invalidPostgresServer',
+                (database) => {
+                    return database.name && !['azure_maintenance', 'azure_sys'].includes(database.name) ? new PostgresDatabaseTreeItem(this, database.name) : undefined;
+                },
+                (database) => database.name
+            );
+        } else {
+            const config = await getClientConfig(this, postgresDefaultDatabase);
```

Comment:
It won't matter in the current case where we are displaying all the databases under an account with or without database name being provided in the connection string. Earlier the specific database name was needed to manage account and non-account scenarios.

Reply:
Actually, for this particular line, we need to use postgresDefaultDatabase; cannot use any other databases. To query all the databases listed under a server we can only do it using 'postgres' database.

Reply:
I was going with the first option of displaying only the account with attaching connectionString, similar to cosmosdb behavior. This release will no longer support attaching database specific connectionString. We need the '/database' in the id for both the behaviors to work.

Reply:
> Actually, for this particular line, we need to use postgresDefaultDatabase; cannot use any other databases. To query all the databases listed under a server we can only do it using 'postgres' database.

We don't need to query the databases, we already know which database they want from the connection string. 

> We need the '/database' in the id for both the behaviors to work.

No we don't. We only need it for a specific edge case bug if the user attaches both at the same time. Mongo attached accounts have supported database-specific connection strings for a while, even with that bug, and no one has complained.

Reply:
I see you are talking about adding database-specific code, I assume you meant modifying account-specific connection string code....
If the bug is not a common occurence then I don't see why we can't have both connection strings supported. The new commit supports database-specific connection strings as well.

Reply:
Yeah I don't think the bug is common. You're welcome to fix it - I just ask that it's part of a separate issue/PR, because I'm not sure changing the id like that is the best way to fix it. Also, ideally you would fix the bug for both PostgreSQL and Mongo at the same time